### DISCUSSION TITLE
Signing Issue in BSC wallet

### DISCUSSION THREAD

**Discussion by mohit-zone** [WalletConnect/walletconnect-monorepo#2624] [Developer Support]
Not getting any response on web after approving the sign request in bsc(binance smart chain) wallet on mobile app(Binance andriod app).I have used web3Modal (Ethereum-wagmi) react integration.

`import React from 'react';
import {
  EthereumClient,
  w3mConnectors,
  w3mProvider,
} from '@web3modal/ethereum';
import { Web3Modal } from '@web3modal/react';
import { configureChains, createConfig, WagmiConfig } from 'wagmi';
import { bsc, bscTestnet } from 'wagmi/chains';

const chains = [bsc, bscTestnet];
const projectId: string = 'projectId';

const { publicClient } = configureChains(chains, [w3mProvider({ projectId })]);
const wagmiConfig = createConfig({
  autoConnect: true,
  connectors: w3mConnectors({ projectId, version: 1, chains }),
  publicClient,
});
const ethereumClient = new EthereumClient(wagmiConfig, chains);

const Web3ModalProvider = ({ children }: any) => {
  return (
    <>
      <WagmiConfig config={wagmiConfig}>{children}</WagmiConfig>
      <Web3Modal projectId={projectId} ethereumClient={ethereumClient} />
    </>
  );
};

export default Web3ModalProvider;
`
import { bscTestnet } from '@wagmi/core/chains';
import { useAccount } from 'wagmi';
import { signMessage, getWalletClient } from '@wagmi/core';

const { isConnected } = useAccount();

useEffect(() => {
   if (isConnected) {
     walletConnect();
    }
  }, [isConnected]);

const { open, setDefaultChain } = useWeb3Modal();

const connectBSCFromMobile=async()=>{
setDefaultChain(bscTestnet);
await open();
}
const walletConnect = async () => {
    try {
      const account = getAccount();
      console.log('account', account);
      const walletClient = await getWalletClient();
      console.log('walletClient', walletClient);
     
     let signature;
    console.log('going to generate signature>>>>>>');
      signature = await walletClient?.signMessage({
        account: walletAddress,
        message: message,
      });
      console.log('signature', signature);
    } catch (err) {
      console.error(err);
      toast.error(`Failed to connect to ${selectedWallet}`);

    }
  };
  
  Also i have used signMessage hook of wagmi/core same issue with that also not getting any response on web after approving request on bsc wallet from mobile

**Comment by glitch-txs** [comment]
hey are you still facing this? could you share a minimal reproducible example repo?

**Reply by mohit-zone** [reply]
yes it is working fine if i am scanning the QR by using Metamask mobile app but when i am scanning the QR by using Binance DEfi wallet scanner then it got stuck in signing ,there is no response on web after approving the signing request on mobile app.I have tried signMessage hook of wagmi/core and walletClient but nothing got worked
signature = await walletClient?.signMessage({
    account: walletAddress,
    message: message,
  });

**Reply by glitch-txs** [reply]
might be an issue on Binance wallet's end, were you able to get in touch with them?

**Comment by finessevanes** [comment]
@mohit-zone do you have a repo we could test with?
